### DISCUSSION TITLE
bug with `getCategories` query in the graphql interface?

### DISCUSSION THREAD

**Discussion by shc023** [swellstores/community#546] [Help] [answered]
I'm trying to use the `where` param in a `getCategories` query to get a list of "sub categories" under a parent category. But when I did
`{where: { "parent.slug": { $eq: "parent"}}}` it did not retrieve anything. I thought it should've worked because in the Swell doc for Product queries it had an example using `where` with `'attributes.material'` as the look-up key.

I did test with just `{where: { "slug": { $eq: "blah"}}}` (with no `parent.` and that worked fine and found a thing, so I'm pretty sure my project's setup correctly.

Similarly for `getCategories`, I tried passing in a `sort` param and wasn't able to get it to work as expected. I tried using `sort: "name asc"`, `sort: "name:asc"`, `sort: "name ASC"` etc. and nothing changed any sort ordering. Is it documented somewhere how to use these params?

**Comment by logeshswell** [comment, marked as answer]
1. Fetching the categories by parent/child is a limitation in the current graphQL implementation.
  a. You can either fetch category by id or slug value.
2. `sort` param in `categories()` is not working as expected. Let me look deeper into this issue and get back to you if I get some updates.
